iroh-dns-server/src/store/signed_packets.rs

```diff
@@ -1,20 +1,145 @@
-use std::{path::Path, sync::Arc};
+use std::{path::Path, result, time::Duration};
 
 use anyhow::{Context, Result};
 use iroh_metrics::inc;
 use pkarr::SignedPacket;
 use redb::{backends::InMemoryBackend, Database, ReadableTable, TableDefinition};
+use tokio::sync::{mpsc, oneshot};
+use tokio_util::sync::CancellationToken;
 use tracing::info;
 
 use crate::{metrics::Metrics, util::PublicKeyBytes};
 
 pub type SignedPacketsKey = [u8; 32];
 const SIGNED_PACKETS_TABLE: TableDefinition<&SignedPacketsKey, &[u8]> =
     TableDefinition::new("signed-packets-1");
+const MAX_BATCH_SIZE: usize = 1024 * 64;
+const MAX_BATCH_TIME: Duration = Duration::from_secs(1);
 
 #[derive(Debug)]
 pub struct SignedPacketStore {
-    db: Arc<Database>,
+    send: mpsc::Sender<Message>,
+    cancel: CancellationToken,
+    thread: Option<std::thread::JoinHandle<()>>,
+}
+
+impl Drop for SignedPacketStore {
+    fn drop(&mut self) {
+        // cancel the actor
+        self.cancel.cancel();
+        // join the thread. This is important so that Drop implementations that
+        // are called from the actor thread can complete before we return.
+        if let Some(thread) = self.thread.take() {
+            let _ = thread.join();
+        }
+    }
+}
```

Comment:
Hm. I worry that this blocking `Drop` implementation will wreak havoc when `SignedPacketStore` is dropped from within a async fn running in tokio.
Can somebody remind me again what the risks there are? Is there no real risk to that?

Reply:
Ah yes, we are basically performing blocking io on the tokio thread, if that happens. But I think blocking the tokio runtime for some time in an emergency is still vastly preferable to not running redb::Database::drop and leaving the db files in an inconsistent state that requires a costly rebuild on restart.

We can add an async shutdown function if we want to give people the option to cleanly shut down the store, but I wanted to keep this minimal.

Reply:
If the worst thing that happens is that we're blocking the tokio runtime for a little bit (and it's just slower during that time), then this is totally fine for me.